casper/src/test/scala/coop/rchain/casper/MultiParenCasperFinalizationSpec.scala

```diff
@@ -19,24 +22,36 @@ class MultiParentCasperFinalizationSpec extends FlatSpec with Matchers with Insp
 
   implicit val timeEff = new LogicalTime[Effect]
 
-  private val (otherSk, otherPk)          = Ed25519.newKeyPair
-  private val (validatorKeys, validators) = (1 to 4).map(_ => Ed25519.newKeyPair).unzip
-  private val (ethPivKeys, ethPubKeys)    = (1 to 4).map(_ => Secp256k1.newKeyPair).unzip
+  private val (validatorKeys, validatorPks) = (1 to 4).map(_ => Ed25519.newKeyPair).unzip
+  private val (_, ethPubKeys) = (1 to 4).map(_ => Secp256k1.newKeyPair).unzip
   private val ethAddresses =
     ethPubKeys.map(pk => "0x" + Base16.encode(Keccak256.hash(pk.bytes.drop(1)).takeRight(20)))
-  private val wallets     = ethAddresses.map(addr => PreWallet(addr, BigInt(10001)))
-  private val bonds       = createBonds(validators)
-  private val minimumBond = 100L
-  private val genesis =
-    buildGenesis(wallets, bonds, minimumBond, Long.MaxValue, true, 0L)
+  private val validators = createBonds(validatorPks).map(Validator.tupled).toSeq
+
+  private val genesisParameters =
+    Genesis(
+      shardId = "MultiParentCasperFinalizationSpec",
+      timestamp = 0L,
+      wallets = ethAddresses.map(PreWallet(_, BigInt(10001))),
+      proofOfStake =
+        ProofOfStake(minimumBond = 0L, maximumBond = Long.MaxValue, validators = validators),
+      faucet = true,
+      genesisPk = Ed25519.newKeyPair._2,
+      vaults = Traverse[List]
+        .traverse(validatorPks.toList)(RevAddress.fromPublicKey)
+        .get
+        .map(Vault(_, 1000L)),
+      supply = Long.MaxValue
```

Comment:
this snippet has been duplicated in every casper test. Please extract a method from that and use it instead, in a separate `fixup! $ORIGINAL_COMMIT_MESSAGE` commit

Reply:
You're correct that the snippet is duplicated. However, some of the constructor actuals are used in various places of the tests. For example, the value `wallets` is used in `genesisParameters` in `MultiParentCasperBondingSpec`, but it's also used later in the tests. We need a better way to pass in the `Genesis` argument (and perhaps additional context information) to each test before we abstract the block construction into a single function. Something like `... in effectTest { genesisParameters => testBody }`. Once we have that, I agree with you 😄. It that function would just take in the amount of each kind of user we want and generate a `Genesis` object.

Reply:
> some of the constructor actuals are used in various places of the tests.

then extract them as values in the test and pass them as parameter to the extracted method

> We need a better way to pass in the Genesis argument (and perhaps additional context information) to each test before we abstract the block construction into a single function. Something like ... in effectTest { genesisParameters => testBody }

either this, or what I've written above. Just do it :) "There wasn't a method extracted upfront" is not an excuse for introducing duplication :smiling_imp: